packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -131,30 +131,96 @@ storage:
       contents:
         inline: |
           #!/bin/bash -xe
+          # Create a RAID 0 from extra disks to be used for persistent container storage.
 
-          function persist_data_raid() {
-            local raid_config=$(mdadm --detail --scan)
-            echo "$${raid_config}" | tee -a /etc/mdadm.conf
-            # Don't mount if no RAID is configured.
-            [ -z "$${raid_config}" ] && return 0
+          function create_data_raid() {
+            local major_numbers="$1"
+            # Filtering by disk type, -1 to not filter, 1 for HDDs and 0 for SSDs
+            local disk_type="$2"
+            # RAID device path, starting with /dev/
+            local device_path="$3"
+            local setup_fs_on_raid="$4"
 
-            cat /etc/mdadm.conf | while read line; do
-              # Make mount persistent across reboots.
-              local devname=$(echo "$${line}" | awk '{print $2}')
+            # Select disks for RAID sorted by size, filter by disk type and
+            # ignore the disks which are mounted (e.g., where Linux is installed)
+            local disks=$(lsblk -lnpd -x size -o path,rota -I "$${major_numbers}" \
+            | (
+            while IFS= read -r line; do
+              local drive=$(echo "$line" | awk '{print $1}')
+              local rota=$(echo "$line" | awk '{print $2}')
+              if [ $$disk_type != -1 ] && [ $$disk_type != $$rota ]; then
+                continue
+              fi
+              local mountpoints=$(lsblk -ln -o mountpoint "$drive")
+              if [[ -z "$mountpoints" ]]; then
+                echo "$line"
+              fi
+            done) | awk '{x=$1 " " x;} END{print x}')
```

Comment:
I think this was here before, so no need to change it.

But something like this maybe is more straigforward to read? Maybe this is only more clear to me because I don't remember much awk syntax :D
```
awk  '{ print $1 }' | tr '\n' ' '
```

Reply:
Yes, good idea!